### DISCUSSION TITLE
Replacing docker with podman?

### DISCUSSION THREAD

**Discussion by kisamoto** [caprover/caprover#1451] [Ideas]
Hi all,

I've been playing with Caprover locally - big fan. 

Is it possible or is there a discussion about replacing docker with rootless podman? I believe the APIs should be similar/compatible but wondered if any research has gone into this already.

Thanks

**Comment by sinistersnare** [comment]
It seems unlikely. CapRover is built with Docker Swarm. Podman has docker-compose capability now, but it seems explicitly unwilling to implement swarm capabilities. 

[Source](https://www.redhat.com/sysadmin/podman-docker-compose), the caveat section at the end.

**Comment by pedrosimao** [comment]
That would be a great upgrade! Using podman kubernetes instead of docker-swarm!

**Reply by kisamoto** [reply]
Having looked a bit closer at this I think it would be quite a project to swap docker-swarm with something like k8s (or even k3s). 

Perhaps I can try hacking something together if I find some time...

**Reply by pedrosimao** [reply]
I am currently looking at kubero.dev.
The project seems quite close to Caprover, but it uses kubernetes.
But I feel like Caprover has a more friendly interface and some extra options for begginers. Maybe the projects could be complimentary, and help each other?

**Reply by pedrosimao** [reply]
I think you could use kuber-cli to achieve kubernetes on caprover without too much work.

**Reply by githubsaturn** [reply]
>  k8s (or even k3s)

The main problem with these platforms is the overkill complexity and resource consumption. At their idle state, they still consume couple of hundred MB of memory.... which is absolutely beyond the acceptable limit for CapRover as most of our user base are on machines <= 2gb of RAM.

**Reply by petoknm** [reply]
I wouldn't mind the lack of swarm/horizontal scaling, I would just like to avoid running Docker on my server. Podman has been my default choice to run containers for some time now and now I found out about Caprover and would love to set it up, but if I have to install docker that puts me off a bit.
